static/js/lib/channels_test.js

```diff
@@ -32,11 +35,36 @@ describe("Channel utils", () => {
     })
   })
 
-  it("isModerator should return true for a moderator user", () => {
-    assert.isTrue(isModerator(makeModerators("username"), "username"))
+  describe("isModerator", () => {
+    it("should return true for a moderator user", () => {
+      assert.isTrue(isModerator(makeModerators("username"), "username"))
+    })
+
+    it("should return false for a regular user", () => {
+      assert.isFalse(isModerator(makeModerators(), "username"))
+    })
   })
 
-  it("isModerator should return false for a regular user", () => {
-    assert.isFalse(isModerator(makeModerators(), "username"))
+  describe("userCanPost", () => {
+    [
+      [CHANNEL_TYPE_PUBLIC, false, false, true],
+      [CHANNEL_TYPE_RESTRICTED, false, false, false],
+      [CHANNEL_TYPE_PRIVATE, false, false, false],
+      [CHANNEL_TYPE_PRIVATE, true, false, true],
+      [CHANNEL_TYPE_PRIVATE, false, true, true]
```

Comment:
should we also test `CHANNEL_TYPE_RESTRICTED` with `isModerator` and `isContributor` set to `true` here?

Reply:
Sure